handler/http/server_test.go

```diff
@@ -53,24 +53,33 @@ func validateEndpoint(endpoint string, round float64) error {
 		return fmt.Errorf("unexpected status: %v", resp.StatusCode)
 	}
 
+	return validateBodyFormat(resp.Body, round)
+}
+
+func validateBodyFormat(respBody io.Reader, round float64) error {
 	body := make(map[string]interface{})
-	if err := json.NewDecoder(resp.Body).Decode(&body); err != nil {
+	if err := json.NewDecoder(respBody).Decode(&body); err != nil {
 		return err
 	}
-	if err := resp.Body.Close(); err != nil {
-		return err
+	if len(body) != 4 && len(body) != 3 {
+		return fmt.Errorf("beacon formatting expects 3 or 4 fields for the beacon")
+	}
+	if rep, ok := body["round"].(float64); !ok || rep != round {
```

Comment:
though a float in js land, it should never seem like a float by the time it reaches here (I appreciate this is just test code though)